### PR TITLE
frontend: Add support for If and IfExpr

### PR DESCRIPTION
**Pull request by georgemitenkov**
This PR adds support for handling Python `if` and `if` expressions.

### CODE REVIEW

**Review by superlopuh — APPROVED** [review]

File: tests/filecheck/frontend/dialects/scf.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            pass
+        elif c:
+            pass
+
+    # CHECK:      %{{.*}} : !i1 = symref.fetch() ["symbol" = @cond]
+    # CHECK-NEXT: scf.if(%{{.*}} : !i1) {
+    # CHECK-NEXT:   scf.yield()
+    # CHECK-NEXT: } {
+    # CHECK-NEXT:   scf.yield()
+    # CHECK-NEXT: }
```
Comment (webmiche) [inline tests/filecheck/frontend/dialects/scf.py:62]: Not sure you followed the discussion in #727 but it seems that the else regions in MLIR can be empty, if they are optional. I am fine with the conversion you are proposing, just wanted to shoutout the MLIR supports a "more native" version of a single-handed if then what you seem to encode here.

Comment (webmiche) [reply tests/filecheck/frontend/dialects/scf.py:62]: I guess the python AST works with these empty regions, so maybe it's not worth the effort of changing/analyzing the structure

**Review by webmiche — APPROVED** [review]
LGTM :+1:

**Review by math-fehr — APPROVED** [review]
Perfect! Thanks

**Pull request merged**